app/models/lookup_keys/puppetclass_lookup_key.rb

```diff
@@ -9,6 +9,7 @@ class PuppetclassLookupKey < LookupKey
   after_validation :reset_override_params, :if => ->(key) { key.override_changed? && !key.override? }
 
   scoped_search :relation => :param_classes, :on => :name, :rename => :puppetclass, :aliases => [:puppetclass_name], :complete_value => true
+  scoped_search :relation => :environments, :on => :name, :rename => :environment, :complete_value => true
```

Comment:
Should this (and possibly the previous one as well) be set to `:only_explicit`? I'm not sure if using free text to search for class or environment is common, and in a large setup this may have a possible performance effect.